ouroboros-network/src/Ouroboros/Network/PeerSelection/PeerStateActions.hs

```diff
@@ -669,10 +679,8 @@ withPeerStateActions PeerStateActionsArguments {
 
           Just (WithSomeProtocolTemperature (WithHot MiniProtocolError{})) -> do
             -- current `pchPeerStatus` must be 'HotPeer'
-            state <- atomically $ do
-              peerState <- readTVar pchPeerStatus
-              _  <- updateUnlessCoolingOrCold pchPeerStatus PeerCooling
-              return peerState
+            state <- atomically do
+              readTVar pchPeerStatus <* updateUnlessCoolingOrCold pchPeerStatus PeerCooling
```

Comment:
I'd live the original, monadic expressions are easier to read (or we're just more used to them).